### ISSUE TITLE
Bug: Website ESQuery doesn't work for typescript nodes

### ISSUE THREAD

**Issue by entropitor** [typescript-eslint/typescript-eslint#8896]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&showAST=es&fileType=.tsx&esQuery=N4IgZglgNgLgpgJxALhAFQMpoJ4Ac4CCAdkQPYwCGMEpRIANCAM5xRwDGMpSyoMecFCAgATOEWqREDEADcKUAK6DUmHPmJlK1WiAC%2BeoA&code=FAYw9gdgzgLgBLATgLgTRBLCBzOBeOAcgBYAmQoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const str: string = '42'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
  }
}
```

### Expected Result

I expected to find the node when looking for it using the esquery query `TSTypeAnnotation`

### Actual Result

I get an empty list

### Additional Info

Probably we need to pass the visitorKeys as options to match (https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website/src/components/ast/ASTViewer.tsx#L27)

### Versions

/

**Comment by bradzacher** [comment]
It appears to be just on existing node types and works fine for new node types?
https://typescript-eslint.io/play/#ts=5.4.3&showAST=es&fileType=.tsx&esQuery=N4IgZglgNgLgpgJxALhAFQMoZgiA7AcwGk4BPAdwHsEATEAGhAGc4o4BjGalUGUgBzgoQEGnDwwIkRAxAA3AIZQArkNSZsuQiQrU6AX31A&code=C4TwDgpgBMULxQM7AE4EsB2BzA3AKAGMB7DZJVALnPW3igHIAWAJnrwDMBXDA4NEqOyJEAFAEMUWKshpYAlNNSYsUAN5QAvniA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false
